reducers/plugins/index.js

```diff
@@ -5,6 +5,44 @@ import {combineReducers} from 'redux';
 
 import {ActionTypes} from 'utils/constants.jsx';
 
+function addMenuItem(menu, data) {
+    if (menu.id === data.parentId) {
+        menu.subMenu = [...menu.subMenu, data];
+    } else if (menu.subMenu) {
+        menu.subMenu = menu.subMenu.map((item) => {
+            const subMenu = {...item};
+            return addMenuItem(subMenu, data);
+        });
+    }
+    return menu;
+}
+
+function buildSubMenu(state, action) {
```

Comment:
without having to pass the full state, i wonder if we can just pass the array of components for `PostDropdownMenu`. Same for action, we need `action` here only to get access to the components, i think it's better to pass `action.data` here. This helps this function to remain agnostic to how data is structured in state/action, only relevant pieces of information is passed in

Reply:
done.  receives the array of components and action.data